packages/@react-spectrum/s2/src/CardView.tsx

```diff
@@ -551,11 +551,13 @@ function CardView<T extends object>(props: CardViewProps<T>, ref: DOMRef<HTMLDiv
     items: props.items, // TODO: ideally this would be the collection. items won't exist for static collections, or those using <Collection>
     onLoadMore: props.onLoadMore
   }, domRef);
+
+  let ctx = useMemo(() => ({size, variant}), [size, variant]);
```

Comment:
Unrelated to the original bug, just something I did while debugging. Avoids re-rendering every card if CardView re-renders but these props don't change.